### DISCUSSION TITLE
How to get global variable in resumed function.

### DISCUSSION THREAD

**Discussion by wisdom1972** [BenLauwens/SimJulia.jl#130] [Q&A] [answered]
**Describe the feature you'd like to have**

- for support team, we can simulate the Human resource, but I want to get the charge for every service. 
- support engineer is resource, we must request, and release after the function occupy the resource. 
- in the function. we want to record what is our charge, for traffic charge, airline charge. 
-

**Cite scientific papers related to the feature/algorithm**

**If possible, sketch out an implementation strategy**

**Comment by Krastanov** [comment]
Can you describe what you have tried and potentially what issues you are getting? A minimal example will be pretty helpful.

There is also the standard non-answer as well -- do not use global variables, they are evil ;)

**Comment by Krastanov** [comment, marked as answer]
Here is a quick generic example (basically you can use a global like in a usual function)

```
julia> using ResumableFunctions

julia> const mygloblist = [];

julia> @resumable function f()
           while true
               a = length(mygloblist)
               push!(mygloblist, a)
               @yield a
               a > 10 && break
           end
       end
f (generic function with 1 method)

julia> collect(f())
12-element Vector{Any}:
  0
  1
  2
  3
  4
  5
  6
  7
  8
  9
 10
 11

julia> mygloblist
12-element Vector{Any}:
  0
  1
  2
  3
  4
  5
  6
  7
  8
  9
 10
 11
```

I will marked this as resolved, but do not hesitate to reopen it if you have additional quesitons
